### ISSUE TITLE
Monsters dropping items when fighting 2 of them at the same time.

### ISSUE THREAD

**Issue by GoogleCodeExporter** [alistairreilly/andors-trail#56]
```
What steps will reproduce the problem?
1. Try to fight two monsters at the same time.

What is the expected output? What do you see instead?
If both of them drop items, they appear only on the tile where the monster
you attacked first was. If monster drops an item, it should be on the same tile 
where monster was.

What version of the product are you using? On what operating system?
v0.6.6, Android 2.1
```

Original issue reported on code.google.com by `rafal.wa...@gmail.com` on 3 Dec 2010 at 12:29

**Comment by GoogleCodeExporter** [comment]
```
This is now fixed. Thank you for reporting it.

There may have been an issue here before. Loot bags were previously only 
created first when the combat was finished. The program kept a list of what 
monsters had been killed in this fight, and should have their loot bags created 
when the combat ends. In order to display a single dialog box with the loot 
items, only one loot bag was created (at the location of the first monster). I 
think this also might have had a side-effect in that the program might not have 
created loot bags for some killed monsters, in particular if you restarted the 
game while in a combat session, since the "killed monsters" list is not 
persisted.

Now the loot bag is created as soon as the monster dies, and the program 
instead keeps a list of what loot bags should be picked up when the combat 
ends. This list is not persisted, but that's not a problem. If you fight two 
monsters, kill the first and then save while in combat, the first loot bag will 
still be there for you to pick up later (which it wouldnt have been before).

This also required an additional change: you now get exp as soon as you kill a 
monster, and not only when combat ends. This way, if you kill one monster, but 
continue fighting another that you get killed by, you will still get the exp 
for the first monster.

Hopefully this fixes a lot of issues with supposed missing items (from bosses 
in particular).

Thank you again rafal.watorski for reporting it.
```

Original comment by `oskar.wi...@gmail.com` on 13 Jan 2011 at 9:28
- Changed state: **Ready**
- Added labels: **Milestone-0.6.8**

**Comment by GoogleCodeExporter** [comment]
```
Before adding a loot bag to the list, there should be a check for pickup 
setting and only add it if the setting says "show loot". If experience can be 
instant, then so can the loot (its not very realistic that our hero has time to 
pickup loot during a fight).
This should resolve issue 109.
```

Original comment by `kims...@gmail.com` on 13 Jan 2011 at 2:42

**Comment by GoogleCodeExporter** [comment]
```
Released in v0.6.8 . Closing.
```

Original comment by `oskar.wi...@gmail.com` on 24 Feb 2011 at 9:32
- Changed state: **Fixed**
